### PR TITLE
Croatian translations

### PR DESCRIPTION
**Pull request by milotype**
Please generate images for:
- /img/card/home-hr.png
- /img/card/generator-hr.png
- /img/logo-osobnipodaci.png

Also, please (if possible) consider to generally "automate" the process of **localizing** links, subdirectories and file names. For instance:
- "email": "contact@datenanfragen.de",
- "url": "https://www.datarequests.org/verein",
- "logo": "https://www.datenanfragen.de/img/logo-datenanfragen-ev.png"

Croatian version would be:
- "email": "kontakt@osobnipodaci.org"
- "url": "https://www.osobnipodaci.org/udruga"
- "logo": "https://www.osobnipodaci.org/img/logo-osobnipodaci.png"

I'm not a programmer, but maybe such data could be automated by using some scripting with Weblate's translation platform ... ??

Because of the above issue, I did not add/translate the subdirectory https://github.com/datenanfragen/website/tree/master/content/en/verein

### CODE REVIEW

**Comment by baltpeter** [comment]
> Please generate images for:

The logo image should already be there. As for the other two, can you give me the translations for the text on those (from the typewriter text on the homepage, "You have a right to privacy" is probably "Imaš pravo na privatnost."; but unless I'm missing something, I don't have a translation for "Easily generate GDPR requests.").

> Also, please (if possible) consider to generally "automate" the process of **localizing** links, subdirectories and file names. For instance:

I definitely agree that the manual process we currently have for those isn't great. Unfortunately, I am not really sure how we could improve this. Our localization is unfortunately glued together over various different libraries and frameworks.

* As for the emails, we don't localize those and sending an email to <anything>@osobnipodaci.org will actually not work and result in an error. That is intentional as no-one on our team speaks Croatian and we thus wouldn't be able to respond to those emails.
* The subdirectories are unfortunately pretty difficult, something some our other translators are also (understandably) disappointed in:
	- The actual folders in the repository need to be named the same across all translations, otherwise Hugo will not realize that they are translated versions of the same content.
	  
      To translate the URL, we need to change the `slug` property of the individual page and then also set the `aliases` property in order for the English paths to still work and redirect to the translated path.
    - As for the `verein` directory, I don't think it's possible to translate that, I'm afraid. That's why all of our language versions use `verein` there.
    - For some pages, like `thanks` and `suggest` can unfortunately not be translated for technical reasons as that would break the parameters we are passing to them.
* As for the logo, `logo-datenanfragen-ev.png` should never be localized as that is the (legal) name of the association. If you meant `logo-datenanfragen-ev.png`, there is already the `hugo.logo` translation or are you talking about something else?

File: content/hr/open-source.md [hunk 1]
```diff
@@ -5,23 +5,129 @@
 	"heading": "<span style='text-align: center; font-family: monospace;'>osobnipodaci.org <span class='color-red-600' title='loves'>❤</span> otvoreni kod</span>"
 }
 
+<img id="open-source-humaaan" class="top-left-humaaan" style="margin-top: -70px;" src="/img/humaaans/open-source.svg" alt="">
+
+Open Source is at the very core of datarequests.org. We have designed the project from the ground up to be as open as possible.
```
Comment (baltpeter) [inline content/hr/open-source.md:10]:
```suggestion
```

File: content/hr/open-source.md [hunk 2]
```diff
@@ -5,23 +5,129 @@
...
-## Contributors
+## Doprinositelji
 
-A huge thanks to everyone contributing to datarequests.org! The project wouldn't be possible without you.
+Zahvaljujemo svima koji su doprinijeli stranicama datarequests.org! Projekt ne bi bio moguć bez vas.
 
 <div class="box box-info" style="white-space: pre;">{{< authors >}}</div>
 
-If you have also contributed but your name isn't on the list yet, we invite you to add it to the [`AUTHORS` file](https://github.com/datenanfragen/website/blob/master/AUTHORS).
+Ako si i ti doprinio/doprinijela projektu, ali tvoje ime još nije na popisu, pozivamo te da ga dodaš u datoteku [`AUTORI`](https://github.com/datenanfragen/website/blob/master/AUTHORS).
```
Comment (baltpeter) [inline content/hr/open-source.md:121]:
```suggestion
Ako si i ti doprinio/doprinijela projektu, ali tvoje ime još nije na popisu, pozivamo te da ga dodaš u datoteku [`AUTHORS`](https://github.com/datenanfragen/website/blob/master/AUTHORS).
```

**Review by baltpeter — COMMENTED** [review]
Thank you, I have left some minor comments.

**Comment by milotype** [comment]
> The logo image should already be there. 

Yes, you are right! Sorry!

> As for the other two, can you give me the translations for the text on those ...

/img/card/home-hr.png
You have a right to privacy.
Imaš pravo na privatnost.

/img/card/generator-hr.png
Easily generate GDPR requests.
Generiraj GDPR zahtjeve jednostavno.

> I definitely agree that the manual process we currently have for those isn't great. Unfortunately, I am not really sure how we could improve this. Our localization is unfortunately glued together over various different libraries and frameworks.

I thought this will be problamatic ... :-( 

>     * As for the emails, we don't localize those and sending an email to @osobnipodaci.org will actually not work and result in an error. That is intentional as no-one on our team speaks Croatian and we thus wouldn't be able to respond to those emails.
IMO, you could use the `@osobnipodaci.org` and other specific (language) domain names for thata matter, as the domain extensions _do not insist/do not directly imply_ the usage a specific language whatsoever. Instead use an "informational text" about used languages for communication. Don't hide such information in a "tooltip", but add a readable text instead. Something like: **("Please contact us in English or German, otherwise we might be unable to understand your request.")**
P.S. I do not insist on this ... please consider it simply as an suggestion/feature request ...

>     * The subdirectories are unfortunately pretty difficult, something some our other translators are also (understandably) disappointed in:
As far as I can tell, you define subdirectories/files and links to HTML files with CSS-IDs at Weblate, like
```
- /verein/become-a-member
- /otvoreni-kod#contributors
```
IMO, you have two choices:

1. let translators transalte those strings (which means, the subdirectories/CSS ids would/could be "translated"
2. until you decide upon the issue, maybe you can specify such strings at Weblate as "read-only", so they don't get translated. PLEASE let me know, because I "translated" such strings at Weblate ...

>       * The actual folders in the repository need to be named the same across all translations, otherwise Hugo will not realize that they are translated versions of the same content.
>         To translate the URL, we need to change the `slug` property of the individual page and then also set the `aliases` property in order for the English paths to still work and redirect to the translated path.
>       * As for the `verein` directory, I don't think it's possible to translate that, I'm afraid. That's why all of our language versions use `verein` there.
>       * For some pages, like `thanks` and `suggest` can unfortunately not be translated for technical reasons as that would break the parameters we are passing to them.
>       * But it might be possible to make it unnecessary to add separate translation strings for the URLs in the footer. That's something I have wanted to look into for a while.

Ideally, you wouldn't need to care about folders/slugs/directories/CSS-IDs ...

>     * As for the logo, `logo-datenanfragen-ev.png` should never be localized as that is the (legal) name of the association. If you meant `logo-datenanfragen-ev.png`, there is already the `hugo.logo` translation or are you talking about something else?

You are right. No need to change, as it is the name of the association.

Bottom line:
do whatever you find usefull or whatever is possible. I'm not a programmer. Let me just know about those above mentioned strings about subdirectories/files and links to HTML files with CSS-IDs at Weblate, so things don't break ...

Milo

**Comment by baltpeter** [comment]
> > As for the other two, can you give me the translations for the text on those ...
> 
> /img/card/home-hr.png
> You have a right to privacy.
> Imaš pravo na privatnost.
> 
> /img/card/generator-hr.png
> Easily generate GDPR requests.
> Generiraj GDPR zahtjeve jednostavno.

Thanks! Do you want me commit the images directly on your branch or upload them here so you can do that?

> IMO, you could use the `@osobnipodaci.org` and other specific (language) domain names for thata matter, as the domain extensions _do not insist/do not directly imply_ the usage a specific language whatsoever. Instead use an "informational text" about used languages for communication. Don't hide such information in a "tooltip", but add a readable text instead. Something like: **("Please contact us in English or German, otherwise we might be unable to understand your request.")**
> P.S. I do not insist on this ... please consider it simply as an suggestion/feature request ...

You're right that it's better to be explicit about this. I agree that adding a hint is useful. Regardless, I am not so sure about the email address for other domains. Setting that up wouldn't be a huge amount of work but it's another moving part… I'll open a new issue (#579) for discussion. Let's see, what the others think.

> Ideally, you wouldn't need to care about folders/slugs/directories/CSS-IDs ...

I have looked into this and now found a much better solution for the footer that completely eliminates the need for the `hugo.footer-*-url` translations. Instead, with the new implementation, Hugo will figure out the correct (translated) URL and we only need to provide the "default" English URL. My new implementation will also automatically fall back to the English version if no translated one exists.

Additionally, we can also use this new implementation in the blog posts. I'll create a new shortcode and you'll then be able to have all the mentioned benefits by simply providing the URL to the English version.

I'll have this implementation ready soon. We'll then have to wait for a review by @zner0L as the changes are fairly "dangerous". Once the PR is merged, I'll update your latest Weblate update #578 accordingly and then we can merge that as well.

> Bottom line:
> do whatever you find usefull or whatever is possible. I'm not a programmer. Let me just know about those above mentioned strings about subdirectories/files and links to HTML files with CSS-IDs at Weblate, so things don't break ...

Will do. And don't worry, we even review translation changes quite carefully, so there's a very low risk of things breaking. :) But of course we want to prevent unnecessary work.

**Comment by milotype** [comment]
> > /img/card/home-hr.png
> > You have a right to privacy.
> > Imaš pravo na privatnost.
> > /img/card/generator-hr.png
> > Easily generate GDPR requests.
> > Generiraj GDPR zahtjeve jednostavno.
> 
> Thanks! Do you want me commit the images directly on your branch or upload them here so you can do that?

If possible, please commit/change it yourself directly ...

> > IMO, you could use the `@osobnipodaci.org` and other specific (language) domain names for thata matter, as the domain extensions _do not insist/do not directly imply_ the usage a specific language whatsoever. Instead use an "informational text" about used languages for communication. Don't hide such information in a "tooltip", but add a readable text instead. Something like: **("Please contact us in English or German, otherwise we might be unable to understand your request.")**
> > P.S. I do not insist on this ... please consider it simply as an suggestion/feature request ...
> 
> You're right that it's better to be explicit about this. I agree that adding a hint is useful. Regardless, I am not so sure about the email address for other domains. Setting that up wouldn't be a huge amount of work but it's another moving part… I'll open a new issue (#579) for discussion. Let's see, what the others think.

OK!
Also, please check the possibility of automatically opening the "localized" version of the website (depending upon the browser's language preferences/settings) when a user enters any of the base URL names in the browsers URL address field (no matter which "translated" site name may be specified - datarequests, datenanfragen, osobnipodaci, demandetesdonnees, pedidodedados, solicituddedatos, or any future translated name). I'm sure this will be difficult ... ;-(

> I have looked into this and now found a much better solution for the footer that completely eliminates the need for the `hugo.footer-*-url` translations. Instead, with the new implementation, Hugo will figure out the correct (translated) URL and we only need to provide the "default" English URL. My new implementation will also automatically fall back to the English version if no translated one exists.
> 
> Additionally, we can also use this new implementation in the blog posts. I'll create a new shortcode and you'll then be able to have all the mentioned benefits by simply providing the URL to the English version.
> 
> I'll have this implementation ready soon. We'll then have to wait for a review by @zner0L as the changes are fairly "dangerous". Once the PR is merged, I'll update your latest Weblate update #578 accordingly and then we can merge that as well.

OK!

> Will do. And don't worry, we even review translation changes quite carefully, so there's a very low risk of things breaking. :) But of course we want to prevent unnecessary work.

OK!

**Comment by baltpeter** [comment]
> If possible, please commit/change it yourself directly ...

Of course, done.

> Also, please check the possibility of automatically opening the "localized" version of the website (depending upon the browser's language preferences/settings) when a user enters any of the base URL names in the browsers URL address field (no matter which "translated" site name may be specified - datarequests, datenanfragen, osobnipodaci, demandetesdonnees, pedidodedados, solicituddedatos, or any future translated name). I'm sure this will be difficult ... ;-(

That's actually something we decided against. We find automatic language redirects a lot more annoying than helpful. But I agree that the discoverability of the other language versions could be better. If you have an idea (design-wise, the implementation should actually not be too difficult) for an unobtrusive way of introducing those to users visiting the "wrong" language version, we can gladly discuss that in a new issue.

---

I have also now finished the new implementation for the links, the PR is open in #580.

Do you want to continue working on this PR or is it done? In the latter case, if you don't mind, I could take over, merge the changes from #578 into here, fix the build problems and then we could merge it as-is. That way, we don't have to wait for the review of #580 (I don't know how long that will take…).

**Comment by milotype** [comment]
> I have also now finished the new implementation for the links, the PR is open in #580.

Thank you.

> Do you want to continue working on this PR or is it done? In the latter case, if you don't mind, I could take over, merge the changes from #578 into here, fix the build problems and then we could merge it as-is. That way, we don't have to wait for the review of #580 (I don't know how long that will take…).

Yes, you can close this issue and proceed as you sugggested ...

**Review by baltpeter — APPROVED** [review]
Thank you very much, @milotype!
